src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,56 @@
+package duke.tasks;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+
+import duke.DukeException;
+
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+    protected String byString;
+
+    /**
+     * Constructor for the Deadline object, a subclass of Task
+     * @param description String containing the description of the task
+     * @param by String setting the deadline for this task
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        this.byString = by;
+        DateTimeFormatter patternWithOptional = new DateTimeFormatterBuilder()
```

Comment:
Code can be kept shorter if the date formatting can be moved to the Parser class.